src/Symfony/Component/Notifier/Bridge/LineBot/LineBotTransport.php

```diff
@@ -0,0 +1,99 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\LineBot;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Yi-Jyun Pan <me@pan93.com>
+ */
+final class LineBotTransport extends AbstractTransport
+{
+    protected const HOST = 'api.line.me';
+
+    public function __construct(
+        #[\SensitiveParameter] private readonly string $channelAccessToken,
+        private readonly string $receiver,
+        ?HttpClientInterface $client = null,
+        ?EventDispatcherInterface $dispatcher = null,
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+        if (null === $this->client) {
+            throw new \LogicException('The LineBot transport cannot send a message without an HTTP client.');
+        }
+
+        $response = $this->client->request(
+            'POST',
+            "https://{$this->getEndpoint()}/v2/bot/message/push",
+            [
+                'headers' => [
+                    'Authorization' => "Bearer $this->channelAccessToken",
+                ],
+                'json' => [
+                    'to' => $this->receiver,
+                    'messages' => [
+                        [
+                            'type' => 'text',
+                            'text' => $message->getSubject(),
+                        ],
+                    ],
+                ],
+            ],
+        );
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (\Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote LINE server.', $response, 0, $e);
+        }
+
+        if (200 !== $statusCode) {
+            $originalContent = $message->getSubject();
+
+            $result = $response->toArray(false) ?: ['message' => ''];
+            \assert(
+                isset($result['message']) && \is_string($result['message']),
+                "A LINE error response should contain a 'message' string key."
+            );
+
+            $errorMessage = trim($result['message']);
+
+            throw new TransportException("Unable to post the LINE message: \"$originalContent\" ($statusCode: \"$errorMessage\").", $response);
```

Comment:
Can we use sprintf here please? also no need for an extra `$errorMessage` variable